### PR TITLE
fix(backend-proxy-middleware): Move handling of bsp to generateProxyMiddlewareOptions

### PR DESCRIPTION
**Pull request by zdravko-georgiev**
Fixes https://github.com/SAP/open-ux-tools/issues/541

### CODE REVIEW

**Review by tobiasqueck — COMMENTED** [review]
Change looks straight forward but your test coverage is unacceptable ;)

**Comment by zdravko-georgiev** [comment]
@tobiasqueck @ullasholla please take a look at https://github.com/SAP/open-ux-tools/pull/542/commits/b412b09e57f25bec25320cf69eccd64c63c6bdd8. I removed `JSON.parse(...)`, because I am getting a parse error when trying to use the proxy middleware with steampunk, `system.serviceKeys` is already a JSON object.

File: packages/backend-proxy-middleware/test/base/proxy.test.ts [hunk 1]
```diff
@@ -367,6 +372,24 @@ describe('proxy', () => {
...
+            const answers = {
+                username: '~user',
+                password: '~password'
+            };
+            mockIsAppStudio.mockReturnValue(false);
+            mockPrompt.mockResolvedValue({ ...answers, authNeeded: true });
+            const options = await generateProxyMiddlewareOptions(backend);
+            expect(options.pathRewrite).toBeDefined();
+            expect(options.router).toBeDefined();
+            expect(options.auth).toBeDefined();
```
Comment (ullasholla) [inline packages/backend-proxy-middleware/test/base/proxy.test.ts:391]: Is it possible to check for more specific values without a deep coupling to the implementation. I'm wondering if just checking for being defined will give us false positives

Comment (zdravko-georgiev) [reply packages/backend-proxy-middleware/test/base/proxy.test.ts:391]: Yes, done with https://github.com/SAP/open-ux-tools/pull/542/commits/2bd3133185713e7db094aff63065da19dd507a94

**Comment by zdravko-georgiev** [comment]
Thanks @ullasholla, I incorporated your feedback. Please review again.

**Review by tobiasqueck — APPROVED** [review]

**Review by ullasholla — APPROVED** [review]

**Pull request merged**